src/khepri_machine.erl

```diff
@@ -667,16 +667,18 @@ process_query_response(
     {error, TimedOut};
 process_query_response(
   StoreId, _RaServer, true = _IsLeader, QueryFun, QueryType, Timeout,
-  {error, noproc})
-  when QueryType =/= local andalso ?HAS_TIME_LEFT(Timeout) ->
+  {error, Reason})
+  when QueryType =/= local andalso ?HAS_TIME_LEFT(Timeout) andalso
+       (Reason == noproc orelse Reason == nodedown) ->
```

Comment:
The same should be applied to `process_sync_command/3` with an associated testcase.